CRM/Core/Form.php

```diff
@@ -359,10 +359,12 @@ public function registerRules() {
    *
    * @return HTML_QuickForm_Element
    *   Could be an error object
+   *
+   * @throws \CRM_Core_Exception
    */
   public function &add(
     $type, $name, $label = '',
-    $attributes = '', $required = FALSE, $extra = NULL
+    $attributes = [], $required = FALSE, $extra = NULL
```

Comment:
@seamuslee001 I think this makes sense for the test fails... it just gets otherwise converted & it's pretty nasty defaulting an array to a string